packages/office-ui-fabric-react/src/components/Keytip/KeytipHost.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { BaseComponent, IRenderComponent } from '../../Utilities';
+import { IKeytipProps } from './Keytip.types';
+import { getNativeKeytipProps, registerKeytip, unregisterKeytip, updateKeytip } from '../../utilities/keytips';
+
+export interface IKeytipHostProps {
+  keytipProps?: IKeytipProps;
+  ariaDescribedBy?: string;
+}
+
+/**
+ * A small element to help the target component correctly read out its aria-describedby for its Keytip
+ *
+ * @export
+ * @class KeytipHost
+ * @extends {BaseComponent<IKeytipProps, {}}>}
+ */
+export class KeytipHost extends BaseComponent<IKeytipHostProps & IRenderComponent<{}>, {}> {
+  private _uniqueId: string;
+
+  public componentDidMount() {
+    // Register Keytip in KeytipManager
+    if (this.props.keytipProps) {
+      this._uniqueId = registerKeytip({ ...this.props.keytipProps });
+    }
+  }
+
+  public componentWillUnmount() {
+    // Unregister Keytip in KeytipManager
+    this.props.keytipProps && unregisterKeytip({ ...this.props.keytipProps }, this._uniqueId);
+  }
+
+  public componentDidUpdate() {
+    // Update Keytip in KeytipManager
+    this.props.keytipProps && updateKeytip({ ...this.props.keytipProps }, this._uniqueId);
+  }
+
+  public render(): JSX.Element {
+    const { children, keytipProps, ariaDescribedBy } = this.props;
+    let nativeKeytipProps: any = {};
+    if (keytipProps) {
+      nativeKeytipProps = getNativeKeytipProps(keytipProps);
+      if (ariaDescribedBy) {
```

Comment:
can you move this ariaDesribedByCheck to the upper if block?

Reply:
this.props.ariaDescribedBy is optional, and is just there to prepend to the one we generate in getNativeKeytipProps. We should still generate the nativeKeytipProps regardless if it's been set. However, I could refactor this so that getNativeKeytipProps takes in the aria-describedBy and can prepend it there